python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -743,25 +739,36 @@ class RandomForestClassifier(Base):
             ndarray, cuda array interface compliant array like CuPy
         y : NumPy
            Dense vector (int) of shape (n_samples, 1)
-
+        algo : string name of the algo from (from algo_t enum)
+               This is optional and required only while performing the
+               predict operation on the GPU.
+               'NAIVE' - simple inference using shared memory
+               'TREE_REORG' - similar to naive but trees rearranged to be more
+                              coalescing-friendly
+               'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
+                                    multiple rows per thread block
+        threshold : float
+                    threshold is used to for classification
+                    This is optional and required only while performing the
+                    predict operation on the GPU.
+        num_classes : integer
+                      number of different classes present in the dataset
         Returns
         -------
         float
            Accuracy of the model [0.0 - 1.0]
         """
         cdef uintptr_t X_ptr, y_ptr
-        X_m, X_ptr, n_rows, n_cols, _ = \
-            input_to_dev_array(X, order='C')
-        y_m, y_ptr, _, _, _ = input_to_dev_array(y)
+        y_m, y_ptr, n_rows, _, y_dtype = input_to_dev_array(y)
```

Comment:
don't you need to check dtype or dimensions here?

Reply:
the dimension gets checked inside the predict function, i will add a check for dtype in it as well.